src/components/sidebar/sidebar.pw.tsx

```diff
@@ -0,0 +1,435 @@
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponent aria-labelledby={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarPreview(page)).toHaveAttribute(
+      "aria-labelledby",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  ([
+    [SIDEBAR_SIZES[0], SIDEBAR_SIZES_CSS["extra-small"]],
+    [SIDEBAR_SIZES[1], SIDEBAR_SIZES_CSS.small],
+    [SIDEBAR_SIZES[2], SIDEBAR_SIZES_CSS["medium-small"]],
+    [SIDEBAR_SIZES[3], SIDEBAR_SIZES_CSS.medium],
+    [SIDEBAR_SIZES[4], SIDEBAR_SIZES_CSS["medium-large"]],
+    [SIDEBAR_SIZES[5], SIDEBAR_SIZES_CSS.large],
+    [SIDEBAR_SIZES[6], SIDEBAR_SIZES_CSS["extra-large"]],
+  ] as [SidebarProps["size"], string][]).forEach(([size, width]) => {
+    test(`check size is ${size}`, async ({ mount, page }) => {
+      await mount(<SidebarComponent size={size} />);
+
+      await assertCssValueIsApproximately(
+        sidebarPreview(page),
+        "width",
+        parseInt(width)
+      );
+    });
+  });
+
+  test("check component has expected header", async ({ mount, page }) => {
+    await mount(<SidebarComponentWithHeading />);
+
+    const sidebarHeaderComponent = getComponent(page, "sidebar-header");
+
+    await expect(sidebarHeaderComponent).toBeVisible();
+
+    const sidebarPreviewHeading = sidebarPreview(page).locator("h3").nth(0);
+    await expect(sidebarPreviewHeading).toHaveText("Sidebar Header");
+  });
+
+  test("check component has role", async ({ mount, page }) => {
+    await mount(<SidebarComponent role={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarPreview(page)).toHaveAttribute(
+      "role",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  test("check component has expected data-element", async ({ mount, page }) => {
+    await mount(<SidebarComponent data-element={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarComponent(page)).toHaveAttribute(
+      "data-element",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  test("check component has expected data-role", async ({ mount, page }) => {
+    await mount(<SidebarComponent data-role={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarComponent(page)).toHaveAttribute(
+      "data-role",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  ([true, false] as const).forEach((boolVal) => {
+    test(`verify visibility of sidebar component when disableEscKey is ${boolVal} and Escape key is pressed`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponentWithOnCancel disableEscKey={boolVal} />);
+
+      const sidebarPreviewElement = sidebarPreview(page);
+      await sidebarPreviewElement.press("Escape");
+      await waitForAnimationEnd(sidebarPreviewElement);
+
+      if (boolVal) {
+        await expect(sidebarPreviewElement).toBeInViewport();
+      } else {
+        await expect(sidebarPreviewElement).not.toBeInViewport();
+      }
+    });
+  });
+
+  ([true, false] as const).forEach((boolVal) => {
+    test(`check component with open prop set to ${boolVal}`, async ({
```

Comment:
**suggestion** (non-blocking): as above, I'd prefer to call this something like `open` or maybe `isOpen`, or similar - rather than `boolVal`